tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,273 @@
+        """
+        Setup operations for the test file.
+        """
+        LOGGER.info("STARTED: Setup Module operations.")
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.csm_user = CMN_CFG["csm"]["csm_admin_user"]["username"]
+        cls.csm_passwd = CMN_CFG["csm"]["csm_admin_user"]["password"]
+        cls.username = []
+        cls.password = []
+        cls.node_master_list = []
+        cls.node_worker_list = []
+        cls.ha_obj = HAK8s()
+        cls.deploy_lc_obj = ProvDeployK8sCortxLib()
+        cls.s3_clean = cls.test_prefix = cls.deply_name = cls.repl_num = cls.test_io_prefix = None
+        cls.def_replica = cls.del_users = cls.header = None
+        cls.mgnt_ops = ManagementOPs()
+        cls.system_random = secrets.SystemRandom()
+        cls.test_dir_path = os.path.join(TEST_DATA_FOLDER, "HATestMultipartUpload")
+        cls.csm_obj = csm_api_factory("rest")
+        cls.csm_soft_delay = HA_CFG["common_params"]["15sec_delay"]
+        cls.num_users = HA_CFG["s3_operation_data"]["iam_users"] + 50
+
+        for node in range(cls.num_nodes):
+            cls.host = CMN_CFG["nodes"][node]["hostname"]
+            cls.username.append(CMN_CFG["nodes"][node]["username"])
+            cls.password.append(CMN_CFG["nodes"][node]["password"])
+            if CMN_CFG["nodes"][node]["node_type"] == "master":
+                cls.node_master_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+            else:
+                cls.node_worker_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+
+        cls.rest_obj = S3AccountOperations()
+
+    def setup_method(self):
+        """
+        This function will be invoked prior to each test case.
+        """
+        LOGGER.info("STARTED: Setup Operations")
+        self.s3_clean = dict()
+        LOGGER.info("Check the overall status of the cluster.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        if not resp[0]:
+            resp = self.ha_obj.restart_cluster(self.node_master_list[0])
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Cluster status is online.")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        set_type, set_name = self.node_master_list[0].get_set_type_name(
+            pod_name=pod_list[0])
+        LOGGER.info("Got set_type = %s and set_name = %s.", set_type, set_name)
+        resp = self.node_master_list[0].get_num_replicas(set_type, set_name)
+        assert_utils.assert_true(resp[0], resp)
+        self.def_replica = int(resp[1])
+        self.repl_num = int(len(self.node_worker_list) / 2 + 1)
```

Comment:
If default replicas are same as that of we are calculating here, we should not again run create_replicas call in test.
We should add one flag for that here.

Reply:
As per design, max can go to 2. This will change depending on how many nodes. Keeping as is it.